lib/neofiles.rb

```diff
@@ -57,6 +57,19 @@ def resized_image_dimensions(image_file, width, height, resize_options)
     # image fits into requested dimensions, no resizing will occur
     return image_file_width, image_file_height if image_file_width <= width && image_file_height <= height
 
+    in_aspect   = 1.0 * image_file_width / image_file_height
+    out_aspect  = 1.0 * width / height
+
+    if in_aspect > out_aspect
+      # If image is more "flat", width will always be equal to the requested width,
+      # and the height will be less than the requested height
+      height = nil
+    else
+      # If input image is more "stretched" vertically or is equal to output aspect ratio,
+      # the height is equal to the requested height, and the width is less than or equal to requested width
```

Comment:
# If input image is more "stretched" vertically or its aspect ratio is equal to output aspect ratio,
# the output height will be equal to the requested height, and the output width will be less than or equal to the requested width